java/app/src/main/java/com/google/android/gms/samples/wallet/CheckoutActivity.java

```diff
@@ -192,53 +222,44 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
    * requested information, such as billing and shipping address.
    *
    * @param paymentData A response object returned by Google after a payer approves payment.
-   * @see <a
-   *     href="https://developers.google.com/pay/api/android/reference/object#PaymentData">Payment
-   *     Data</a>
+   * @see <a href="https://developers.google.com/pay/api/android/reference/
+   *      object#PaymentData">PaymentData</a>
    */
   private void handlePaymentSuccess(PaymentData paymentData) {
-    String paymentInformation = paymentData.toJson();
 
     // Token will be null if PaymentDataRequest was not constructed using fromJson(String).
-    if (paymentInformation == null) {
-      return;
-    }
-    JSONObject paymentMethodData;
+    final String paymentInfo = paymentData.toJson();
+    if (paymentInfo == null) return;
 
     try {
-      paymentMethodData = new JSONObject(paymentInformation).getJSONObject("paymentMethodData");
+      JSONObject paymentMethodData = new JSONObject(paymentInfo).getJSONObject("paymentMethodData");
       // If the gateway is set to "example", no payment information is returned - instead, the
       // token will only consist of "examplePaymentMethodToken".
-      if (paymentMethodData
-              .getJSONObject("tokenizationData")
-              .getString("type")
-              .equals("PAYMENT_GATEWAY")
-          && paymentMethodData
-              .getJSONObject("tokenizationData")
-              .getString("token")
-              .equals("examplePaymentMethodToken")) {
-        AlertDialog alertDialog =
-            new AlertDialog.Builder(this)
+
+      final JSONObject tokenizationData = paymentMethodData.getJSONObject("tokenizationData");
+      final String tokenizationType = tokenizationData.getString("type");
+      final String token = tokenizationData.getString("token");
+
+      if ("PAYMENT_GATEWAY".equals(tokenizationType) && "examplePaymentMethodToken".equals(token)) {
```

Comment:
I'm not sure if there's a convention for this, but they look backwards to me, eg:

`if (tokenizationType.equals("PAYMENT_GATEWAY") && token.equals("examplePaymentMethodToken")) {`

Reply:
This prevents a `NullPointerException` f any of `tokenizationType` or `token` are `null`. Putting the definite value first makes the operation work resiliently no matter what's on the other side (eg.: if `tokenizationType` is `null`, the condition will test `false`)

Reply:
That's clever and makes sense, thanks